### PR TITLE
Create snapshot of Prometheus data and upload to S3

### PR DESCRIPTION
**Pull request by aleksbykov**

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -3203,16 +3204,44 @@ def download_monitor_data(self):
...
+                snapshot_archive = self.get_snapshot(node)
+                self.log.debug('Snapshot local path: {}'.format(snapshot_archive))
+
+                return S3Storage.upload_file(snapshot_archive)
             except Exception, details:
                 self.log.error('Error downloading prometheus data dir: %s',
                                str(details))
         return ""
 
+    def get_snapshot(self, node):
```
Comment (amoskong) [inline sdcm/cluster.py]: Currently we use phantomjs to get snapshot, can we use the new graceful method to replace phantomjs?

Comment (bentsi) [reply sdcm/cluster.py]: @amoskong we can. If you have any other method that you know please add a task in NEW column in Trello and we will consider replacing phantomJS

Comment (bentsi) [reply sdcm/cluster.py]: BTW this method is not related to screenshot that we take. There is a difference between the words `snapshot` and `screenshot`. Snapshots usually used in DB, filesystem context, while screenshot is used to capture screen.
To make things clear, Alex, please rename `get_snapshot` to `get_prometheus_snapshot`

File: sdcm/cluster.py [hunk 2]
```diff
@@ -3203,16 +3204,44 @@ def download_monitor_data(self):
...
+        archive_snapshot_name = '/tmp/{}.tar.gz'.format(Setup.test_id())
+        result = node.remoter.run('cd {}; tar -czvf {} .'.format(snapshot_dir,
+                                                                 archive_snapshot_name),
+                                  ignore_status=True)
+        if result.exit_status > 0:
+            self.log.warning('Unable to create archive {}'.format(result))
+            return ""
+
+        try:
+            local_snapshot_data_dir = os.path.join(node.logdir, "snapshot_data")
```
Comment (bentsi) [inline sdcm/cluster.py]: `snapshot_data` -> `prometheuse_data`

File: sdcm/cluster.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/cluster.py]: Renamed

File: sdcm/cluster.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/cluster.py]: done

**Comment by bentsi** [comment]
I am going to merge this tomorrow, @fruch please review.

**Pull request merged**